woocommerce/woocommerce
# Remove all schema checking code from Code Analyzer.

### Changes proposed in this Pull Request:

It has been decided for now to remove schema checking functionality from Code Analyzer for a number of reasons:

1. It was unreliable. We have to spin up a wp-env environment which should be OK, but it hangs or breaks sometimes and we haven't been able to determine why

2. Due to this unreliable nature (and it being very slow too) no one was using the tool. At least for now we decided we'd rather the tool do less, but be much much faster and more reliable.

As a result we'll need teams to report schema changes manually going forward but that's fine since it wasn't really working before anyway.

I also made some minor improvements to types and error handling, and removed schema checking from release post tool as well.

### How to test the changes in this Pull Request:

Run some sample commands as suggested in the README and confirm they still have outputs and they don't crash, produce errors or warnings of any kind. Note these commands should be run from the code-analyzer directory.

1. `pnpm run analyzer -- lint "release/6.8" "6.8.0" -b release/6.7`
2. `pnpm analyzer scan hooks "release/6.8" "release/6.7" --since "6.8.0"`
3. Also run the release post generator as it had some minor changes. Go to `tools/release-posts`
4. Setup GITHUB_ACCESS_TOKEN as described in the README
5. Run `pnpm run release-post release "6.8.0" "6.7.0" --outputOnly`. There should be no errors and the output HTML file should be viewable. (The path to the HTML file will be output in terminal).




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
